docs/2.guide/5.best-practices/security.md

```diff
@@ -0,0 +1,306 @@
+
+### Authentication failures
+
+Authentication and session management are vital to security. Weaknesses may include allowing automated attacks (e.g., credential stuffing, brute force), using default or weak passwords, poor password recovery methods, storing passwords insecurely, lacking proper multi-factor authentication, exposing session IDs in URLs, reusing session IDs after login, and failing to invalidate sessions on logout or inactivity.
+
+A simple example attack scenarion can be:
+
+```
+A user uses a public computer to access an application. Instead of selecting "logout," the user simply closes the browser tab and walks away. An attacker uses the same browser an hour later, and the user is still authenticated.
+```
+
+To protect against such attack, you could implement a proper authentication method. Check out Nuxt Auth Utils [below](#nuxt-auth-utils).
+
+:read-more{title="Identification and Authentication Failures" to="https://owasp.org/Top10/A07_2021-Identification_and_Authentication_Failures/"}
+
+## Content Security Policy
+
+Content Security Policy (CSP) helps prevent unwanted content from being injected/loaded into your webpages. This can mitigate cross-site scripting (XSS) vulnerabilities, clickjacking, formjacking, malicious frames, unwanted trackers, and other web client-side attacks.
+
+:read-more{title="Content Security Policy" to="https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/CSP"}
+
+The CSP header can have the following structure:
+
+```
+Content-Security-Policy: default-src 'self'; img-src 'self' example.com
+```
+
+Currently, a proper Content Security Policy can be enabled by using the Nuxt Security module but we are working on making it part of the core framework. If you are interested in learning more about it, check out [this RFC](https://github.com/nuxt/nuxt/discussions/30404).
+
+Nuxt Security helps you increase the security of your site by enabling Strict CSP support for both SSR and SSG applications.
+
+1. For `SSR` applications, Nuxt Security implements strict CSP via nonces. A one-time cryptographically-generated random nonce is generated at runtime by the server for each request of a page. In this rendering mode, you can also use all security headers in a regular form of a response header (including headers such as Cross Origin Embeder Policy, Cross Origin Opener Policy, etc).
+2. For `SSG` applications, Nuxt Security implements strict CSP via hashes. At static build-time, Nuxt Security computes the SHA hashes of the elements that are allowed to execute on your site. The CSP header value is then delivered as `<meta http-equiv>` tag and you can also implement Subresource Integrity that enables the browser to verify that the static assets that your application is loading have not been altered.
+
+:read-more{title="Strict Content Security Policy" to="https://nuxt-security.vercel.app/advanced/strict-csp"}
+
+
+## Third-party modules
```

Comment:
btw, IMO it is probably not a good idea to mention specifics modules that are not part of the official modules.

This can fracture the community as we're showing a kind of favoritism toward some maintainers/authors.

Reply:
Agree about favoritism but the problem is that there is no official module such as Nuxt Image, Nuxt Scripts, etc. All these modules are community - not official modules and I thought that we should at least recommend something that will help users find answers for their questions or needs.

@danielroe what are your thoughts on that? Both about that and maybe about the general direction of this particular docs. Should we focus on security issues coming from OWASP and how to solve them in Nuxt? Or maybe just about Content Security Policy? But then, we will be missing content about other risks that are equally or even more important. 

I would love to get your feedback on that to better understand how to write this properly :)

Reply:
> the problem is that there is no official module such as Nuxt Image, Nuxt Scripts, etc.

That's an excellent point you made here.
I personnally think that authorizations and auth-utils should remain outside nuxt core and nuxt official modules as there's different ways to implement authorizations/auth. We could always document anti-patterns tho. 
Some solutions provided by nuxt-security should indeed be handled, not by Nuxt but probably by Nitro since things like CSP could be built-in within Nitro.

Let's wait for @danielroe 's opinion :eyes: